### PR TITLE
Add a toggle

### PR DESCRIPTION
**Pull request by oal55**
image

## Summary of changes:
- Adds mantine UI, so that we can build nice looking stuff without much effort.
- Creates blessed functions for google's old fashioned storage-api functions. 
- Adds an enable/disable toggle for the plugin into our popup.
- Content script is now a TS file, and respects the new local setting

### CODE REVIEW

File: src/content.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { getFromLocalStorage } from "./chrome-utils/localStorage";
+
+const localStorageKey = "isExtensionEnabled"; // BAD! maintain keys centrally.
+
+document.addEventListener('mouseup', async function (event) {
+    const isEnabled = await getFromLocalStorage(localStorageKey) as boolean | undefined;
+    const selectedText = window.getSelection()?.toString().trim() ?? "";
+    if (isEnabled !== true) {
+        return;
+    }
```
Comment (oal55) [inline src/content.ts:8]: This file is almost a clean copy/paste. Only added 3-4 lines here to return when the plugin is disabled.

File: src/popup/app.tsx [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
+import { MantineProvider } from '@mantine/core';
+import { SettingsPage } from "./SettingsPage";
 
-const Popup: React.FC<{}> = () => <div>Popup</div>;
+import '@mantine/core/styles.css';
 
 const ROOT_DIV_ID = 'popup-root';
 const rawRootDiv = document.getElementById(ROOT_DIV_ID);
 if (rawRootDiv != null) {
+    chrome.storage.local.get
```
Comment (oal55) [inline src/popup/app.tsx]: Remove

File: src/popup/style.scss [hunk 3]
```diff
@@ -1,3 +1,7 @@
 * {
     margin: 0px;
 }
+
+.main-card {
+    width: 400px;
+}
```
Comment (oal55) [inline src/popup/style.scss]: \n

File: src/popup/SettingsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+export const SettingsPage: React.FC<{}> = () => {
+    return (
+        <Card withBorder shadow="sm" radius="md" className="main-card">
+            <Card.Section withBorder inheritPadding py="xs">
+                <Text fw={500}>Settings</Text>
+            </Card.Section>
+
+            <Text mt="sm" c="dimmed" size="sm">
+                We can populate this nicely with
```
Comment (oal55) [inline src/popup/SettingsPage.tsx]: add an empty space here 🫠

File: src/content.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import { getFromLocalStorage } from "./chrome-utils/localStorage";
+
+const localStorageKey = "isExtensionEnabled"; // BAD! maintain keys centrally.
```
Comment (oal55) [inline src/content.ts]: Pull these out (this one and one in content script.

File: esbuild.js [hunk 6]
```diff
@@ -7,12 +7,15 @@ const filesToStaticallyCopy = [
     "icon.png",
     "popup.html",
     "background.js",
-    "content.js"
+    // "content.js"
```
Comment (oal55) [inline esbuild.js]: Remove this line, and nuke the file

File: package.json [hunk 7]
```diff
@@ -18,9 +18,14 @@
     "chrome-types": "^0.1.263",
     "esbuild": "^0.20.0",
     "esbuild-plugin-copy": "^2.1.1",
-    "esbuild-sass-plugin": "^3.0.0"
+    "esbuild-sass-plugin": "^3.0.0",
+    "postcss": "^8.4.35",
+    "postcss-preset-mantine": "^1.13.0",
+    "postcss-simple-vars": "^7.0.1"
```
Comment (oal55) [inline package.json]: Understand how exactly `mantine` uses `postcss-...` stuff. See if we can get away without depending on these.

They don't affect bundle size, but we're still pulling a bunch of things into node modules o.O

File: popup.html [hunk 8]
```diff
@@ -2,7 +2,9 @@
 <html>
   <head>
     <title>CSV Matching Extension</title>
-    <script src="Popup.js" defer></script>
+    <script src="popup/app.js" defer></script>
+    <link rel="stylesheet" href="popup/app.css">
+    <link rel="stylesheet" href="popup/style.css">
```
Comment (oal55) [inline popup.html]: This file should be placed close to its css/js friends imo. I.e. Same folder

**Review by oal55 — COMMENTED** [review]
Will FLUP on these

File: package.json (hunk 7 above)
Comment (oal55) [reply package.json]: ^ okay looks like they are not _required_, just recommended; and we don't currently use them. Hence removed.

**Review by irmk323 — APPROVED** [review]

**Comment by irmk323** [comment]
Felt a full of sense on summary change comment.. this guy..

**Pull request merged**